### ISSUE TITLE
NPM Token and possible Github_Token for va-forms-system-core

### ISSUE THREAD

**Issue by bjmccotter7192** [department-of-veterans-affairs/va.gov-team#39606]
Hi there, I am working on some refinement to some github actions for the repository "va-forms-system-core" and I need to update the NPM token to the organizations NPM token and I also need a Github Personal Token.

Thanks!

**Comment by bjmccotter7192** [comment]
@oseasmoran73 Yes I am currently using my own NPM Token and would like to use the VA_VSP_BOT_NPM token. The other token is for a github action that required the NPM token and a `secrets.GITHUB_TOKEN`. I am not sure what the standard is for requesting that type of token, so that's why I said possible Github Personal Token. 

Basically I just want to make sure that this repo can use a VA/VSP NPM Token and some sort of VA/VSP Github Action Token so that I can implement future workflows that may require permission to check-in auto generated code.

**Comment by bjmccotter7192** [comment]
I found an example of someone doing a similar action that I would like to implement into the va-forms-system-core:

- name: Semantic Release
        uses: cycjimmy/semantic-release-action@v2
        id: semantic
        env:
          GITHUB_TOKEN: ${{ env.VA_VSP_BOT_GITHUB_TOKEN }}
          
Semantic versioning and release is the prototyped action I have that uses the NPM token and the GITHUB_TOKEN and I believe I need this one above.

**Comment by oseasmoran73** [comment]
Ah okay! I just wanted to make sure that we were on the same page. 

Yeah no worries! Would you happen to know if the team you has an AWS service account (SVC)? This would be used to authenticate and be able to get tokens.

**Comment by bjmccotter7192** [comment]
Pretty sure we don't, I think I may be the only one that has requested AWS access for my team and when I log in I have no access to anything so what would be the best way to get that setup?

**Comment by oseasmoran73** [comment]
The SVC would need to be setup by us. I can take a look into it on our end. What is your team name?

I also updated ticket based off our convo. Feel free to edit/modify if it does not seem accurate

**Comment by bjmccotter7192** [comment]
Perfect so the team name is [`pst-forms-library-team`](https://github.com/orgs/department-of-veterans-affairs/teams/pst-forms-library)

**Comment by bjmccotter7192** [comment]
So for the ticket I was currently working on I didn't end up needing the Github_Token thanks to githubs auto publishing github pages, and we are not planning on pushing any new version to NPM until about the end of Q2 (for a demo).

TLDR - not really a rush but just would like to get this setup so I don't have to worry about tokens in the future.

**Comment by oseasmoran73** [comment]
@bjmccotter7192 It looks like the [Npm bot](https://www.npmjs.com/~va_vsp_bot_npm) is part of the organization. Once PR gets approved, the service account for GHA will be created

We need to add secret/key to the repo, can you provide the URL for the repo as well as make us admins to add those `secrets`?

**Comment by oseasmoran73** [comment]
I edited my previous message as I realized I didn't provide group 😅 , `vsp-operations`

**Comment by bjmccotter7192** [comment]
Alright I got you all added and if you need me to change permissions on that group just let me know

**Comment by oseasmoran73** [comment]
All set! I added `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` to the repo for you to call secret

You will need to authenticate to pull from parameter store to retrieve NPM token such as the following:

```
- name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws-region: us-gov-west-1

      - name: Get NPM Token password
        uses: marvinpinto/action-inject-ssm-secrets@v1.2.1
        with:
          ssm_parameter: /devops/VA_VSP_BOT_NPM_TOKEN
          env_variable_name: VA_VSP_BOT_NPM_TOKEN
```

Which then you can use `${{ env.VA_VSP_BOT_NPM_TOKEN }}` to access token. 

Once you have verified that it is working as intended, can you check off those on acceptance criteria and close out this ticket?

Let me know if you run into any issues :)

**Comment by oseasmoran73** [comment]
I think you can run a [whoami](https://docs.npmjs.com/cli/v7/commands/npm-whoami) to verify you are getting the bot token

**Comment by bjmccotter7192** [comment]
Thank you, I will work this in ASAP and let you know if I have any issues!

**Comment by bjmccotter7192** [comment]
@oseasmoran73 Looks like that worked great! I was able to pull the NPM token and publish with no issues, thank you so much for all your help.
